Add option to save IMS file in a specified location

lguerard opened the pull request:
Fixes #94

lguerard committed:
Add option to save IMS file in a specified location

ehrenfeu commented on src/imcflibs/imagej/misc.py:
```diff
@@ -669,7 +669,7 @@ def locate_latest_imaris(paths_to_check=None):
 
 
-def run_imarisconvert(file_path, pixel_calibration=None):
+def run_imarisconvert(file_path, output_folder=None, pixel_calibration=None):
```
The way this change is done is potentially breaking existing scripts, as they might simply supply parameters in-order (i.e. not using their names) even though the parameter is optional.
By adding it *after* the previously existing parameters, existing calls will still work.

lguerard replied:
I reordered the parameters and docstring info. ✅

ehrenfeu replied:
Fixed in f6804ee6081aad659902e9243655fa38be8e8972 👍🏼

lguerard committed:
feat(misc): reorder parameters in run_imarisconvert function

- Changed the order of parameters in the run_imarisconvert function to improve compatibility.
- Updated docstring to reflect the new parameter order.

ehrenfeu commented on src/imcflibs/imagej/misc.py:
```diff
@@ -680,11 +680,11 @@ def run_imarisconvert(file_path, output_folder=None, pixel_calibration=None):
         Sequence of 3 values (x, y, z) representing voxel dimensions to be set during
         conversion, by default None.
+    output_folder : str, optional
+        Folder where the newly created IMS file will be saved, by default None.
```
What happens if `output_folder` is not specified? Also, would be preferable to have the default defined as `output_folder=""` in order to have type consistency (`str` vs. `NoneType`).

lguerard replied:
I had it that it was using `os.getcwd()` which was quite stupid, so now, if not specified, the file will be created next to the input file.

ehrenfeu replied:
I think we should have this in the docstring, so it's clear for everyone else without having to read up the code.

ehrenfeu replied:
Fixed in ac1e860facc4526b943e1513f62b47b9c25f3e43 and 2c604ed254068bde1965433ed36595a700f3336c 👍🏼

lguerard committed:
fix(run_imarisconvert): set default output folder to file's directory

* Updated the default output folder in run_imarisconvert function to the directory of the input file if not specified.
* This change ensures that converted files are saved in a more intuitive location.

lguerard committed:
feat(run_imarisconvert): enhance output folder handling

- Add notes on output folder behavior in docstring.
- Specify that if no output folder is provided, the IMS file will be saved in the input file's directory.
- Clarify output filename construction from input filename.

lguerard committed:
fix(run_imarisconvert): clarify output folder behavior

- Update docstring to specify that if no output folder is provided, the IMS file will be saved in the same directory as the input file.

ehrenfeu committed:
Merge branch 'devel' into pr/100

ehrenfeu committed:
Docstring formatting / conventions

ehrenfeu committed:
Supply input file name in log messages

ehrenfeu commented:
Needs to be mentioned in changelog: #126
